jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/LCAAlgorithm.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.interfaces;
+
+import org.jgrapht.alg.util.Pair;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Algorithm to compute a <a href="https://en.wikipedia.org/wiki/Lowest_common_ancestor">lowest common ancestor</a>
+ * in a tree, forest or DAG.
+ *
+ * @param <V> vertex the graph vertex type
+ *
+ * @author Alexandru Valeanu
+ */
+public interface LCAAlgorithm<V> {
+
+    /**
+     * Return the LCA of a and b
+     *
+     * @param a the first element to find LCA for
+     * @param b the other element to find the LCA for
+     *
+     * @return the LCA of a and b, or null if there is no LCA.
+     */
+    V getLCA(V a, V b);
+
```

Comment:
Shouldn't we expose getAllLCAs at the interface level as well?  Algorithms are free to throw UnsupportedOperationException if they don't support it.

Reply:
I don't think so. That method is only added to classes that can implement the function more efficiently than the default one (only Tarjan so far)  and to make it easier for the user to get LCAs for a batch of queries.

Reply:
I'm referring to the getAllLCAs in NaiveLCAFinder, which finds all the lowest common ancestors (in the case where there are multiple) instead of just returning one chosen arbitrarily.

Reply:
Sorry, I completely misread it. We can expose and make all implementation except NaiveLCAFinder return the result of getLCA. What do you think?

Reply:
That works, but then it shouldn't have "all" in the name since it's not guaranteed to return a comprehensive set (and as with isomorphism inspectors, each alg doc should specify its own behavior).  How about

getLCA(V, V) : returns one
getLCASet(V, V) : returns many
getBatchLCA(List): batch version, each query returns one
getBatchLCASet(List): batch version, each query returns many

Reply:
I agree. I assume that the batch version is implemented in terms of the non-batch one.